### ISSUE TITLE
Installation question

### ISSUE THREAD

**Issue by devel0** [Atrejoe/MarkdownPreview#2]
I tried from a cmd admin prompt but installation fail

```
2016-07-27  12:24    <DIR>          .
2016-07-27  12:24    <DIR>          ..
2014-11-29  21:10            34,304 Apex.WinForms.dll
2016-06-01  09:34            14,848 MarkdownPreview.dll
2016-06-01  09:34            17,920 MarkdownPreview.pdb
2016-06-01  09:34             3,025 MarkdownPreview.xml
2016-05-27  13:11            54,272 MarkdownSharp.dll
2014-11-30  19:17           172,032 ServerManager.exe
2014-11-30  19:16           384,512 SharpShell.dll
2014-11-30  22:12           347,648 SharpShell.pdb
2014-11-30  22:12           622,788 SharpShell.xml
2014-11-30  19:17           402,432 srm.exe

>**srm install MarkdownPreview.dll**

========================================
SharpShell - Server Registration Manager
========================================

Preparing to install (OS64Bit): MarkDown Preview Handler
**Failed to install and register the server.**
```

Trying from the GUI

ServerManager.exe -> Load ( choose of MarkdownPreview.dll )
![capture](https://cloud.githubusercontent.com/assets/13405008/17172495/acd744ce-53f5-11e6-8777-ac75df1e25d7.JPG)

Clicking on the Server -> Register Server (x64) I got follow

![capture2](https://cloud.githubusercontent.com/assets/13405008/17172555/efd37e82-53f5-11e6-8c86-52fb325f808f.JPG)

Any hint ?
Thanks in advice

**Comment by Atrejoe** [comment]
This is a SharpShell issue I think. It look like the installation tool (which is maintained by @dwmkerr) does not assert the correct registry keys.

Would you mind addressing it here: https://github.com/dwmkerr/sharpshell/issues ?

**Comment by devel0** [comment]
I did it, just now.

**Comment by Atrejoe** [comment]
Have you tried different settings in the GUI? Like registering & installing the 32 bit version?

**Comment by devel0** [comment]
yes, it results the same error.

**Comment by tbone2k-git** [comment]
It's the same here! Bummer! Tried srm.exe and the ServerManager GUI tool with all options to no avail.

It seems srm.exe gives "Failed to install and register the server." when called with "MarkdownPreview.dll" -codebase. It's also quite unclear with which DLL to run srm.exe, since there are 3 of them in your release download. Tried them all, "MarkdownSharp.dll, SharpShell.dll" seem to pass installation at least, not knowing if this is correct or even necessary.

With the ServerManager GUI tool, I get the "MarkdownPreview.dll" file to appear under "Installation" for x86/x64 in the right pane, but not under "Registration", which I guess is required as well.

Thanks!

**Comment by Atrejoe** [comment]
The correct dll would be `MarkdownPreview.dll`, this has a dependency on the other two.

Even on my x64 system, I could only install the x86 version, resulting is:

![image](https://cloud.githubusercontent.com/assets/585091/18388139/b7d26494-76a0-11e6-8907-bf06d0a7a213.png)

**Comment by Atrejoe** [comment]
@tbone2k-git did you try both installing and registering using x86?

![image](https://cloud.githubusercontent.com/assets/585091/18779859/105d3942-817b-11e6-8ce2-c449d9025260.png)

**Comment by tbone2k-git** [comment]
@Atrejoe Thanks for your response! o)
Yes I did. "Install Server" seems to work for both (x86 and x64), but "Register Server" does **_not**_ work. "Register Server" always fails for both versions with "The specified path is invalid". Any ideas appreciated, would really like to use this gem! o)

**Comment by JohnTasler** [comment]
My intuition tells me that srm.exe is a 32-bit app. Because of this, it can't load a 64-bit dll.

**Comment by ewa** [comment]
There seems to still be some issue.  I added a comment on #4 .

**Comment by stanpetit** [comment]
Hi I had trouble installing with same symptoms as described above and at [#4](https://github.com/Atrejoe/MarkdownPreview/issues/4),  
So I checked the ``SharpShell.dll`` versions :   
MarkdownPreview uses SharpShell.dll : 2.2.0.0

Get the proper release of ``ServerManager.zip`` [here](https://github.com/dwmkerr/sharpshell/releases)

It works for my Windows7, register and install as x86, the preview is lagging a bit but I wouldn't be able to do any better !
So usefull ! @Atrejoe :+1: Thanks a lot !

Hope it helps.
Best
Stan

**Comment by Atrejoe** [comment]
That's cool to hear. I just had a fresh Win10 reinstall, so it's time to install this gem again ;).

I may have a go at updating the dependencies and releasing an updated version too.

Thanks for your enthusiastic response @stanpetit !

**Comment by stanpetit** [comment]
Hi @Atrejoe ! I am definitely enthusiastic about  what can make my life easier ! You did that, that's an interesting effort when Markdown is everywhere today ...  thanks a lot !

About  an updated release :   
-  would you check if the preview scrolls alright with the mouse wheel ? It doesn't on my win7.
-  the vertical scroll bar does not scroll the preview smoothly either : it makes jumps on mouse-chick on the handle (not on the whole vertical bar) and makes the whole Explorer display flicker a bit. 
-  More precisely : even with small files, it takes quite many seconds to begin scrolling smoothly downwards, and then it becomes a real mess when scrolling upwards.
-  [edit] and now it does "same as description above", with the reverse down-up direction ...

With that kind of framework, I guess that SharpShell is providing interfaces to implement the event   handlers, maybe even a cache mechanism.

Good luck, 
Best
Stan

**Comment by Atrejoe** [comment]
@stanpetit I have the scrolling issue too, I'm completely unable to scroll (see #3).

I'm about to release a [new version](https://github.com/Atrejoe/MarkdownPreview/releases/tag/v0.4-alpha), upgrading SharpShell (fingers crossed) and replacing the obsolete MarkDown-interpretation engine with Markdig.

**Comment by stanpetit** [comment]
@Atrejoe Hi!
The quite random behavior of the scrolling makes me think of an over-usage in the md interpreter, because it doesn't just forbid to scroll : it seems just too busy to release the action to the user ...
Maybe an event makes it do it over and over.
I would seek this way before to look into a SharpShell issue, because SharpShell seems to do it's job ok : the binding to the preview is done, even if the event loop is weird !
Good luck,
Best
Stan

**Comment by stanpetit** [comment]
@Atrejoe : last reply was late !
I saw your comment 
[at new version](https://github.com/Atrejoe/MarkdownPreview/releases/tag/v0.4-alpha)
and I am convinced that an event or a state forbids the md interpreter to take a rest when the temporary html file is produced. It should be produced once and cached to to some tmp ...

**Comment by Atrejoe** [comment]
@stanpetit let's continue that discussion at #3 
(but I don't think it's a loading/performance issue, when I add a static counter and display it, it behaves nicely)

**Comment by Atrejoe** [comment]
Marked as fixed by release [0.6-alpha](../releases/tag/0.6-alpha)
